migrations/versions/0363_cancelled_by_api_key.py

```diff
@@ -0,0 +1,56 @@
+"""
+Revision ID: 0363_cancelled_by_api_key
+Revises: 0362_broadcast_msg_event
+Create Date: 2022-02-09 14:05:27.750234
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+revision = '0363_cancelled_by_api_key'
+down_revision = '0362_broadcast_msg_event'
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('broadcast_message', sa.Column('created_by_api_key_id', postgresql.UUID(as_uuid=True), nullable=True))
+    op.add_column(
+        'broadcast_message', sa.Column('cancelled_by_api_key_id', postgresql.UUID(as_uuid=True), nullable=True)
+    )
+    op.drop_constraint('broadcast_message_api_key_id_fkey', 'broadcast_message', type_='foreignkey')
+    op.create_foreign_key(
+        'broadcast_message_created_by_api_key_id_fkey',
+        'broadcast_message',
+        'api_keys',
+        ['created_by_api_key_id'],
+        ['id']
+    )
+    op.create_foreign_key(
+        'broadcast_message_cancelled_by_api_key_id_fkey',
+        'broadcast_message',
+        'api_keys',
+        ['cancelled_by_api_key_id'],
+        ['id']
+    )
+    op.get_bind()
+    op.execute("UPDATE broadcast_message SET created_by_api_key_id=api_key_id")  # move data over
+    op.create_check_constraint(
+        "ck_broadcast_message_created_by_not_null",
+        "broadcast_message",
+        "created_by_id is not null or created_by_api_key_id is not null"
+    )
```

Comment:
this constraint cant be created until after we've updated all the code to create with the new way.

or between this and #3452 we'll create messages that will fail this check

Reply:
ahh you are right, I will move it, and copying of the data between columns, to the later migration.

Reply:
ok done now, I moved both data copying and create constraint to the later migration ☺️